sql-native/src/main/java/org/ballerinalang/sql/parameterprocessor/DefaultResultParameterProcessor.java

```diff
@@ -639,24 +656,23 @@ public Object convertTimestampWithTimezone(java.time.OffsetDateTime offsetDateTi
                             zoneMap.put(StringUtils.fromString(
                                     org.ballerinalang.stdlib.time.util.Constants.ZONE_OFFSET_RECORD_MINUTE), 0);
                         }
-
                         if (zoneInfo.get(org.ballerinalang.stdlib.time.util.Constants.ZONE_OFFSET_RECORD_SECOND)
                                 != null) {
                             zoneMap.put(StringUtils.fromString(
                                     org.ballerinalang.stdlib.time.util.Constants.ZONE_OFFSET_RECORD_SECOND),
                                     zoneInfo.get(org.ballerinalang.stdlib.time.util.Constants.ZONE_OFFSET_RECORD_SECOND)
                                             .longValue());
                         }
-
                         zoneMap.freezeDirect();
                         civilMap.put(StringUtils.fromString(
                                 org.ballerinalang.stdlib.time.util.Constants.CIVIL_RECORD_UTC_OFFSET), zoneMap);
                         civilMap.put(StringUtils.fromString(
                                 org.ballerinalang.stdlib.time.util.Constants.CIVIL_RECORD_TIME_ABBREV),
-                                offsetDateTime.getOffset().toString());
+                                StringUtils.fromString(offsetDateTime.getOffset().toString()));
                         return civilMap;
                     } else {
-                        return Utils.createTimeStruct(Timestamp.valueOf(offsetDateTime.toLocalDateTime()).getTime());
+                        throw new ApplicationError("Unsupported Ballerina type:" +
+                            type.getName() + " for SQL Timestamp with timezone Data type.");
```

Comment:
```suggestion
                            type.getName() + " for SQL Timestamp with timezone data type.");
```

Reply:
Updated. Commit id is ba7d30e7b37789ba67ba680c36f81bf3a61a4588